lib/utils.js

```diff
@@ -496,10 +496,24 @@ function isTvOS (platformName) {
   return _.toLower(platformName) === _.toLower(PLATFORM_NAME_TVOS);
 }
 
+/**
+ * Returns true if the urlString is localhost
+ * @param {?string} urlString
+ * @returns {boolean} Return true if the urlString is localhost
+ */
+function isLocalHost (urlString) {
+  try {
+    const {hostname} = new URL(urlString);
```

Comment:
why not url.parse ?

Reply:
Since it moves to https://nodejs.org/api/url.html#url_legacy_url_api, _Stability: 0 - Deprecated: Use the WHATWG URL API instead._

Reply:
https://github.com/appium/appium-base-driver/blob/dc5e5ef3e01cbc581c83f1034f3dac78d114dad8/lib/basedriver/helpers.js#L77

Reply:
ok, lets keep this one if parse is deprecated